ddterm/shell/extension.js

```diff
@@ -140,6 +142,50 @@ class ExtensionDBusInterface {
     }
 }
 
+class DesktopEntry {
+    constructor() {
+        this.target_file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'applications',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+        this.source_file = Gio.File.new_for_path(GLib.build_filenamev(
+            [
+                GLib.get_user_data_dir(),
+                'gnome-shell',
+                'extensions',
+                Me.metadata.uuid,
+                'ddterm',
+                `${APP_ID}.desktop`,
+            ]
+        ));
+    }
+
+    _ensure_install_directory_exists() {
+        var install_directory = this.target_file.get_parent();
+        if (!GLib.file_test(install_directory.get_path(), GLib.FileTest.IS_DIR))
+            install_directory.make_directory_with_parents(null);
+    }
+
+    _is_installed() {
+        return GLib.file_test(this.target_file.get_path(), GLib.FileTest.EXISTS);
```

Comment:
Since `this.target_file` is a `Gio.File`, wouldn't it be easier to call [`.exists()`](https://gjs-docs.gnome.org/gio20~2.0/gio.file#method-query_exists)?

Reply:
I'm using the `OVERWRITE` flag on install and catching errors on uninstall, as suggested, so the _in_installed() function is not needed anymore.